ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -2062,6 +2071,66 @@ - (ORKAnswerFormat *)impliedAnswerFormat {
 @end
 
 
+#pragma mark - ORKConfirmTextAnswerFormat
+
+@implementation ORKConfirmTextAnswerFormat
+
+- (Class)questionResultClass {
+    return [ORKBooleanQuestionResult class];
+}
+
+- (instancetype)initWithOriginalItemIdentifier:(NSString *)originalItemIdentifier {
+    self = [super init];
+    if (self) {
+        _originalItemIdentifier = [originalItemIdentifier copy];
+    }
+    return self;
+}
+
+- (BOOL)isAnswerValid:(id)answer {
+    return [(NSNumber *)answer isEqual:@YES] ? YES : NO;
```

Comment:
Don't necessarily need to check if answer is empty because it is a non-optional form item. But for robustness, I'm going to add the check, just in case if someone wants to use it as an optional item.